apps/sms/views/shared/js/activity_picker.js

```diff
@@ -3,51 +3,36 @@
 (function(exports) {
 'use strict';
 
-function handleActivity(activity, onsuccess, onerror) {
-  // Note: the MozActivity is intentionally constructed in the caller and
-  // passed down to this function in order to make it possible to pass an
-  // inline option to its constructor in the caller for the sake of static
-  // analysis tools.  Please do not change that!
-
-  if (typeof onsuccess === 'function') {
-    activity.onsuccess = onsuccess;
-  }
-
-  activity.onerror = typeof onerror === 'function' ? onerror : function(error) {
-    console.warn('Unhandled error spawning activity; ' + error.message);
-  };
-}
-
 var ActivityPicker = {
-  dial: function ap_call(number, onsuccess, onerror) {
-    handleActivity(new MozActivity({
+  dial: (number) => {
```

Comment:
Using arrow function didn't give any additional benefit here, so I would suggest you could simply follow es6 object Property definitions(https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#Property_definitions)